src/ess.jl

```diff
@@ -255,33 +255,34 @@ function ess_rhat(
 )
     # compute size of matrices (each chain may be split!)
     niter = size(chains, 1) ÷ split_chains
-    nparams = size(chains, 3)
     nchains = split_chains * size(chains, 2)
     ntotal = niter * nchains
+    axes_out = (axes(chains, 3),)
 
     # do not compute estimates if there is only one sample or lag
     maxlag = min(maxlag, niter - 1)
-    maxlag > 0 || return fill(missing, nparams), fill(missing, nparams)
+    if !(maxlag > 0)
+        return similar(chains, Missing, axes_out), similar(chains, Missing, axes_out)
```

Comment:
Do we have any tests for this? And any `@test_inferred` checks - it seems they should fail due to this line?

Reply:
Yes, these lines are tested: https://github.com/TuringLang/MCMCDiagnosticTools.jl/blob/b6e8de9a6ac330c4f7f6cc45185d1e509127fa1f/test/ess.jl#L122-L134.

I added more tests that the array type returned here preserves the final axis and some `@inferred` tests specifying the type union.

Reply:
I agree, this is a good idea.